From be16b7859171456388b411cfba991a988907ef49 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Wed, 19 Jun 2019 11:20:26 +0800
Subject: [PATCH] arm64: dts: rk3399pro: using multicodec sound for rt5651
 codec

We need to export Headphone Jack snd ctl to userspace
on linux platform via multicodec machine driver.

Change-Id: I5664a0b29dfda2ed8cc450a5f0fd388d32dfdd48
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 .../dts/rockchip/rk3399pro-evb-v10-linux.dts  | 26 +++++--------------
 .../dts/rockchip/rk3399pro-evb-v11-linux.dts  | 26 +++++--------------
 2 files changed, 14 insertions(+), 38 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
index 842ba328d9eb..d16e249e9a9d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
@@ -163,25 +163,13 @@
 		};
 	};
 
-	rk809-sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,name = "rockchip,rk809-codec";
-		simple-audio-card,mclk-fs = <256>;
-		simple-audio-card,widgets =
-			"Microphone", "Mic Jack",
-			"Headphone", "Headphone Jack";
-		simple-audio-card,routing =
-			"Mic Jack", "MICBIAS1",
-			"IN1P", "Mic Jack",
-			"Headphone Jack", "HPOL",
-			"Headphone Jack", "HPOR";
-		simple-audio-card,cpu {
-			sound-dai = <&i2s1>;
-		};
-		simple-audio-card,codec {
-			sound-dai = <&rk809_codec>;
-		};
+	rk809_sound: rk809-sound {
+		compatible = "rockchip,multicodecs-card";
+		rockchip,card-name = "rockchip,rk809-codec";
+		rockchip,codec-hp-det;
+		rockchip,mclk-fs = <256>;
+		rockchip,cpu = <&i2s1>;
+		rockchip,codec = <&rk809_codec>;
 	};
 
 	rk_headset: rk-headset {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
index f09124972911..2cc795127bbb 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
@@ -163,25 +163,13 @@
 		};
 	};
 
-	rk809-sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,name = "rockchip,rk809-codec";
-		simple-audio-card,mclk-fs = <256>;
-		simple-audio-card,widgets =
-			"Microphone", "Mic Jack",
-			"Headphone", "Headphone Jack";
-		simple-audio-card,routing =
-			"Mic Jack", "MICBIAS1",
-			"IN1P", "Mic Jack",
-			"Headphone Jack", "HPOL",
-			"Headphone Jack", "HPOR";
-		simple-audio-card,cpu {
-			sound-dai = <&i2s1>;
-		};
-		simple-audio-card,codec {
-			sound-dai = <&rk809_codec>;
-		};
+	rk809_sound: rk809-sound {
+		compatible = "rockchip,multicodecs-card";
+		rockchip,card-name = "rockchip,rk809-codec";
+		rockchip,codec-hp-det;
+		rockchip,mclk-fs = <256>;
+		rockchip,cpu = <&i2s1>;
+		rockchip,codec = <&rk809_codec>;
 	};
 
 	rk_headset: rk-headset {
-- 
2.35.3

